drivers/tty/serdev/pibridge.c

```diff
@@ -0,0 +1,409 @@
+#include <linux/errno.h>
+#include <linux/of.h>
+#include <linux/serdev.h>
+#include <linux/wait.h>
+#include <linux/kfifo.h>
+#include <linux/mutex.h>
+#include <linux/module.h>
+
+#define REV_PI_IO_TIMEOUT           10         // msec
+#define REV_PI_RECV_BUFFER_SIZE     100
+
+#define pb_err(n, fmt, ...) \
+	do {\
+		static int cc = 0;\
+		if (cc < n && pibridge_s ){\
+			dev_err(&pibridge_s->serdev->dev, fmt, ##__VA_ARGS__);\
+			cc ++;\
```

Comment:
Hm, this is called with "n = 25" everywhere.

@zhan-kunbus What's the motivation for limiting these error messages to 25 and dropping all of them once that limit is reached?

Maybe `dev_err_ratelimited()` would be more appropriate?

Reply:
AFAIK those custom functions to print messages are not well received by some maintainers

Reply:
> AFAIK those custom functions to print messages are not well received by some maintainers

Also I am  sure that accessing variables in a macro that have to be defined by the caller (like pibridge_s in pb_err()) is frawned upon. I vote for not printing any messages at all in the serdev module. Instead lets define some sane return values for error conditions and the let the caller decide how to handle these.